### ISSUE TITLE
concourse -  'there is no unique or exclusion constraint matching the ON CONFLICT specification'

### ISSUE THREAD

**Issue by poblin-orange** [orange-cloudfoundry/paas-templates#1224]
### Expected behavior

* As a paas-templates operator 
* In order to operate
* I need to have a concourse up and running

### Observed behavior
 
Context;
- 51.0.5
- bump to concourse 7.3.2

all concourse db access fail with
- web ui:
```
 there is no unique or exclusion constraint matching the ON CONFLICT specification
```

- db logs:
```
2021-06-19 09:19:27.310 GMT: ERROR:  there is no unique or exclusion constraint matching the ON CONFLICT specification
2021-06-19 09:19:27.310 GMT: STATEMENT:  INSERT INTO worker_resource_caches (resource_cache_id,worker_base_resource_type_id) VALUES ($1,$2)
                                ON CONFLICT (resource_cache_id, worker_base_resource_type_id) DO UPDATE SET
                                        resource_cache_id = $3,
                                        worker_base_resource_type_id = $4
                                RETURNING id

```

web log

```
quest","team":"main"}}
{"timestamp":"2021-06-19T09:23:37.940395769Z","level":"info","source":"atc","message":"atc.tracker.notify.run.errored","data":{"build":"check","build_id":57563175,"error":"pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"osb-cmdb-pipeline","resource":"PR-osb-cmdb-to-mirrored-upstream","session":"23.82320.368","team":"main"}}
{"timestamp":"2021-06-19T09:23:37.946256520Z","level":"info","source":"atc","message":"atc.tracker.notify.run.errored","data":{"build":"check","build_id":57563175,"error":"run check: pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"osb-cmdb-pipeline","resource":"PR-osb-cmdb-to-mirrored-upstream","session":"23.82320.368","team":"main"}}
{"timestamp":"2021-06-19T09:23:37.952011816Z","level":"info","source":"atc","message":"atc.tracker.notify.run.finish.errored","data":{"build":"check","build_id":57563175,"error":"run check: pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"osb-cmdb-pipeline","resource":"PR-osb-cmdb-to-mirrored-upstream","session":"23.82320.368.7","team":"main"}}
{"timestamp":"2021-06-19T09:25:31.526001697Z","level":"error","source":"atc","message":"atc.tracker.notify.run.get-step.create.failed-to-initialize-cache","data":{"build":"check","build_id":57564282,"error":"pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"micro-depls-bosh-generated","resource":"concourse-meta","session":"23.82729.129.5.17","step-name":"meta","team":"micro-depls"}}
{"timestamp":"2021-06-19T09:25:31.526599671Z","level":"info","source":"atc","message":"atc.tracker.notify.run.errored","data":{"build":"check","build_id":57564282,"error":"pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"micro-depls-bosh-generated","resource":"concourse-meta","session":"23.82729.129","team":"micro-depls"}}
{"timestamp":"2021-06-19T09:25:31.534583620Z","level":"info","source":"atc","message":"atc.tracker.notify.run.errored","data":{"build":"check","build_id":57564282,"error":"run check: pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"micro-depls-bosh-generated","resource":"concourse-meta","session":"23.82729.129","team":"micro-depls"}}
{"timestamp":"2021-06-19T09:25:31.542044657Z","level":"info","source":"atc","message":"atc.tracker.notify.run.finish.errored","data":{"build":"check","build_id":57564282,"error":"run check: pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"micro-depls-bosh-generated","resource":"concourse-meta","session":"23.82729.129.7","team":"micro-depls"}}
{"timestamp":"2021-06-19T09:25:33.621572753Z","level":"error","source":"atc","message":"atc.tracker.notify.run.get-step.create.failed-to-initialize-cache","data":{"build":"check","build_id":57564286,"error":"pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"master-depls-cached-buildpack-pipeline","resource":"weekday-morning","session":"23.82729.125.5.17","step-name":"cron-resource","team":"main"}}
{"timestamp":"2021-06-19T09:25:33.622360754Z","level":"info","source":"atc","message":"atc.tracker.notify.run.errored","data":{"build":"check","build_id":57564286,"error":"pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"master-depls-cached-buildpack-pipeline","resource":"weekday-morning","session":"23.82729.125","team":"main"}}
{"timestamp":"2021-06-19T09:25:33.628541702Z","level":"info","source":"atc","message":"atc.tracker.notify.run.errored","data":{"build":"check","build_id":57564286,"error":"run check: pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"master-depls-cached-buildpack-pipeline","resource":"weekday-morning","session":"23.82729.125","team":"main"}}
{"timestamp":"2021-06-19T09:25:33.634701460Z","level":"info","source":"atc","message":"atc.tracker.notify.run.finish.errored","data":{"build":"check","build_id":57564286,"error":"run check: pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"master-depls-cached-buildpack-pipeline","resource":"weekday-morning","session":"23.82729.125.7","team":"main"}}
{"timestamp":"2021-06-19T09:25:34.491766727Z","level":"error","source":"atc","message":"atc.tracker.notify.run.get-step.create.failed-to-initialize-cache","data":{"build":"check","build_id":57564264,"error":"pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"control-plane","resource":"concourse-micro-legacy","session":"23.82729.146.5.17","step-name":"concourse-5-pipeline","team":"main"}}
{"timestamp":"2021-06-19T09:25:34.492382956Z","level":"info","source":"atc","message":"atc.tracker.notify.run.errored","data":{"build":"check","build_id":57564264,"error":"pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"control-plane","resource":"concourse-micro-legacy","session":"23.82729.146","team":"main"}}
{"timestamp":"2021-06-19T09:25:34.498347286Z","level":"info","source":"atc","message":"atc.tracker.notify.run.errored","data":{"build":"check","build_id":57564264,"error":"run check: pq: there is no unique or exclusion constraint matching the ON CONFLICT specification","pipeline":"control-plane","resource":"concourse-micro-legacy","session":"23.82729.146","team":"main"}}
{"timestamp":"2021-06-19T09:25:34.504398444Z","level":"info","source":"atc","message":"atc.tracker.notify.run.finish.errored","data":{"build":"check","build_id":57564264,"error":"run check: pq: there is no uniqu
```

### Affected releases

* x.y
* earlier versions

**Comment by o-orand** [comment]
This issue occurred during concourse 7.3.2 upgrade.
During upgrade, we encounter a DB upgrade failure, probably due to slow migration (from 7.2.0 to 7.3.2), so bosh rollback to 7.2.0. Unfortunately, it seems data in DB where still in 7.3.2, so we had an inconsistency.  
To fix, we downloaded bosh manifest, we bumped stemcell to 1.10, updated all refences, bumped concourse to 7.3.2 and manually deploy this updated manifest.
